src/v/model/record_batch_reader.h

```diff
@@ -263,7 +263,7 @@ class record_batch_reader final {
     operator<<(std::ostream& os, const record_batch_reader& r);
 };
 
-using record_batch_reader_opt = ss::optimized_optional<record_batch_reader>;
+using record_batch_data_opt = std::optional<record_batch_reader::data_t>;
```

Comment:
?

Reply:
the commit message doesn't mention these changes at all

Reply:
no need for optimized_optional anymore?

Reply:
There were no references to a record_batch_reader_opt in the source, so it was removed. I don't think you can wrap a ss::circular_buffer in an optimized optional. I got a compiler error. I think this is because theres no overload for operator bool for that type.

Reply:
I'll edit the commit msg